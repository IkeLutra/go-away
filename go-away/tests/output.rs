#![allow(dead_code)]

use insta::assert_snapshot;

use go_away::{TypeMetadata, TypeRegistry};

#[derive(TypeMetadata)]
struct MyData {
    field_one: String,
    nested: Nested,
}

#[derive(TypeMetadata)]
struct Nested {
    #[serde(rename = "some_other_name")]
    a_string: String,
    an_int: i64,
}

#[test]
fn test_struct_output() {
    let mut registry = TypeRegistry::new();
    MyData::metadata(&mut registry);

    assert_snapshot!(go_away::registry_to_output(registry));
}
