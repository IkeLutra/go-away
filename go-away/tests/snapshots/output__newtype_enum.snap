---
source: go-away/tests/output.rs
expression: "go_away::registry_to_output(registry)"

---
type One struct {
	X float64 `json:"x"`
}
type Two struct {
	Y bool `json:"y"`
}
type NewTypeEnum struct {
	*OptionOne
	*OptionTwo
}

func (self *NewTypeEnum) MarshalJSON() ([]byte, error) {
	if self.OptionOne != nil {
		var output map[string]interface{}
		output["k"] = "OptionOne"
		output["c"] = self.OptionOne
		return json.Marshal(output)
	} else 	if self.OptionTwo != nil {
		var output map[string]interface{}
		output["k"] = "OptionTwo"
		output["c"] = self.OptionTwo
		return json.Marshal(output)
	}
}

